### PR TITLE
updtae stream info in server to support seamless failover

### PR DESCRIPTION
**Pull request by yossipapi**

### CODE REVIEW

File: lib/BackendClient.js [hunk 1]
```diff
@@ -265,8 +279,28 @@ var BackendClient = (function(){
...
 
+        var liveEntryServerNode = new kalturaVO.KalturaLiveEntryServerNode();
+        var streamInfoArray = [];
+        _.each(flavorsObjArray, function(flavorInfo) {
+            var KalturaLiveStreamParams = new kalturaVO.KalturaLiveStreamParams();
+            KalturaLiveStreamParams.bitrate = flavorInfo.bandwidth;
+            KalturaLiveStreamParams.flavorId = flavorInfo.name;
+            KalturaLiveStreamParams.width = flavorInfo.resolution[0];
+            KalturaLiveStreamParams.height = flavorInfo.resolution[1];
+
```
Comment (gadyaari) [inline lib/BackendClient.js]: Check if "resolution" even exists for audio only flavors to avoid exceptions.

File: lib/BackendClient.js [hunk 2]
```diff
@@ -235,6 +236,19 @@ var BackendClient = (function(){
         return deferred.promise;
     }
 
+    function updateStreamInfo(entryId, entryServerNodeId, liveEntryServerNode)
+    {
+        var deferred = Q.defer();
+        logger.debug("[%s] Calling updateStreamInfo for entryServerNode id [%s], with stream info array [$j]",entryId,  entryServerNodeId, liveEntryServerNode.streams);
+
+        client.entryServerNode.update(function(result, err, headers) {
```
Comment (gadyaari) [inline lib/BackendClient.js]: Check error and print incase of fail

**Comment by yossipapi** [comment]
@gadyaari  added the changes requested please re-review.

File: lib/kaltura-client-lib/KalturaClientBase.js [hunk 3]
```diff
@@ -384,13 +385,29 @@ function sendRequestHelper(that, options, body, requestIndex, onCompleteCallback
 			for ( var header in response.headers) {
 				headers.push(header + ': ' + response.headers[header]);
 			}
-			that.debug('Headers [' + requestIndex + ']: \n\t' + headers.join('\n\t'));
-			that.debug('Response [' + requestIndex + ']: ' + data);
-			onCompleteCallback(JSON.parse(data), null, headers);
+			This.debug('Headers [' + requestIndex + ']: \n\t' + headers.join('\n\t'));
+			This.debug('Response [' + requestIndex + ']: ' + data);
+			if (This.config.format != KalturaClientBase.KALTURA_SERVICE_FORMAT_JSON){
+				onCompleteCallback(data);
```
Comment (kalturaguy) [inline lib/kaltura-client-lib/KalturaClientBase.js]: didn't we lost the headers passing feature now?

Comment (yossipapi) [inline lib/kaltura-client-lib/KalturaClientBase.js]: Yes we did ... I will restore it back

**Pull request closed without merging**